tests/components/ipp/test_sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Tests for the IPP sensor platform."""
+from datetime import datetime
+
+from asynctest import patch
+
+from homeassistant.components.ipp.const import DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import ATTR_ICON, ATTR_UNIT_OF_MEASUREMENT, UNIT_PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.util import dt as dt_util
+
+from tests.components.ipp import init_integration
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_sensors(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the creation and values of the IPP sensors."""
+    entry = await init_integration(hass, aioclient_mock, skip_setup=True)
+    registry = await hass.helpers.entity_registry.async_get_registry()
+
+    # Pre-create registry entries for disabled by default sensors
+    registry.async_get_or_create(
+        SENSOR_DOMAIN,
+        DOMAIN,
+        "cfe92100-67c4-11d4-a45f-f8d027761251_uptime",
+        suggested_object_id="epson_xp_6000_series_uptime",
+        disabled_by=None,
+    )
+
+    test_time = datetime(2019, 11, 11, 9, 10, 32, tzinfo=dt_util.UTC)
+    with patch("homeassistant.components.ipp.sensor.utcnow", return_value=test_time):
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.epson_xp_6000_series")
+    assert state
+    assert state.attributes.get(ATTR_ICON) == "mdi:printer"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) is None
+
+    state = hass.states.get("sensor.epson_xp_6000_series_black_ink")
+    assert state
+    assert state.attributes.get(ATTR_ICON) == "mdi:water"
+    assert state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) is UNIT_PERCENTAGE
+    assert state.state == "58"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5274251)